src/components/todolist/todolist.tsx

```diff
@@ -0,0 +1,146 @@
+// import { TodoInfo } from '../todoinfo/todoinfo';
+import classNames from 'classnames';
+import { useTodos } from '../../context/todoProvider';
+import { deleteTodo } from '../../api/todos';
+import { ErrorType } from '../../types/Error';
+import { TodoItem } from '../todoItem/todoItem';
+
+export const TodoList = () => {
+  const {
+    visibleTasks, setError, tempTodo, todos,
+    setTodos, deletingTask, setDeletingTask,
+    isAddingTask,
+  } = useTodos();
+
+  const handleDeleteClick = (id: number) => {
+    setError(null);
+
+    setDeletingTask([...deletingTask, id]);
+
+    deleteTodo(id)
+      .then(() => {
+        const filteredTodo = todos.filter(task => task.id !== id);
+
+        setTodos(filteredTodo);
```

Comment:
use prevState here as we discussed during call, this is the same situation
So it would go like:
```
setTodos((currentTodos: Todo[]) => {
    return currentTodos.filter(task => task.id !== id);
});
```